File: services/language-service/src/main/java/dev/animedia/languageservice/application/mapper/LanguageApplicationMapper.java
package dev.animedia.languageservice.application.mapper;

import dev.animedia.languageservice.application.dto.LanguageDto;
import dev.animedia.languageservice.domain.model.Language;
import org.springframework.stereotype.Component;

@Component
public class LanguageApplicationMapper {
    public Language toDomain(LanguageDto dto) {
        return new Language(
            dto.code(),
            dto.name(),
            dto.isActive(),
            dto.isDefault(),
            dto.sortOrder(),
            dto.flagEmoji()
        );
    }

    public LanguageDto toDto(Language language) {
        return new LanguageDto(
            language.getCode(),
            language.getName(),
            language.getIsActive(),
            language.getIsDefault(),
            language.getSortOrder(),
            language.getFlagEmoji()
        );
    }
}

File: services/social-service/src/core/Application/Language/Mapper/LanguageApplicationMapper.php
<?php

declare(strict_types=1);

namespace Core\Application\Language\Mapper;

use Core\Application\Language\DTO\LanguageCommandDto;
use Core\Application\Language\DTO\LanguageResponseDto;
use Core\Domain\Language\Entity\Language;

class LanguageApplicationMapper implements LanguageApplicationMapperInterface
{
    #[\Override]
    public function toLanguage(LanguageCommandDto $languageDto): Language
    {
        return new Language(
            isoCode: $languageDto->languageIsoCode,
            name: $languageDto->name,
            active: $languageDto->isActive
        );
    }

    #[\Override]
    public function toLanguageResponseDto(?Language $language): ?LanguageResponseDto
    {
        if (null === $language) { return null; }
        return new LanguageResponseDto(
            languageIsoCode: $language->isoCode,
            name: $language->name,
            active: $language->active
        );
    }
}
